setup.py

```diff
@@ -20,9 +20,8 @@
 ]
 
 testing_extras = [
-    'coverage',
-    'nose',
-    'nosexcover',
+    'pytest',
+    'pytest-cov',
```

Comment:
It's worth noting that `nose` is used solely as the test runner (discovery).

It is true that the testing tools are for devs, but please remember that `letsencrypt.test` module is intentionally distributed as a part of the core source tree in order to allow any user to run such tests, etc. (background reading for such design decision can be easily googled). On the other hand, I guess that you're suggesting `py.test` not only to allow `assert x == y`-like tests syntax, but also to use `import pytest` and use `pytest.*` functions inside test modules, am I right? If so, then `pytest` would need to be included inside `install_requires` just like in case of `mock` (as opposed to `testing_extras`). Luckily, `pytest` itself seems to be available in Debian stable, and it's rather unnecessary to include problematic `pytest-cov` in `install_requires`.

@jdkasten has a fair aim to support as many operating systems and environments as possible, hence e.g. support for Python 2.6. While you're correct about jessie freeze, note that even squeeze is still alive and marked as LTS! My comment about Debian stable was just an example, but it seems to me that `nose` is much more popular and it's worth checking the support in other OSes before proceeding.